src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder.cs

```diff
@@ -56,17 +56,22 @@ internal static partial class DeclarationFinder
                 var isExactNameSearch = query.Kind == SearchKind.Exact ||
                     (query.Kind == SearchKind.ExactIgnoreCase && !syntaxFacts.IsCaseSensitive);
 
-                // Do a quick syntactic check first using our cheaply built indices.  That will help us avoid creating
-                // a compilation here if it's not necessary.  In the case of an exact name search we can call a special 
-                // overload that quickly uses the direct bloom-filter identifier maps in the index.  If it's nto an 
-                // exact name search, then we will run the query's predicate over every DeclaredSymbolInfo stored in
-                // the doc.
-                var containsSymbol = isExactNameSearch
-                    ? await project.ContainsSymbolsWithNameAsync(query.Name!, cancellationToken).ConfigureAwait(false)
-                    : await project.ContainsSymbolsWithNameAsync(query.GetPredicate(), filter, cancellationToken).ConfigureAwait(false);
-
-                if (!containsSymbol)
-                    return;
+                // This shortcut only works if there cannot be any generated documents. If there are generated documents, then
+                // we'll need to realize the compilation to get them before concluding the symbol doesn't exist.
+                if (!project.State.SourceGenerators.Any())
```

Comment:
this change cannot be made.  Practically all .net6 projects have source generators, so this will cause us to have a perf hit there no matter what.  We discovered just how bad this was with navigate to, and are still spending time trying to claw the performance back.  

My position is that we can only add SG support when we can do so in a manner that comes with no perf impact (or minimal perf impact) in the case that people *have* source generator assemblies, but aren't actually generating anything.  

In other words, if i'm on .net6/.net7 and i'm not taking advantage of SGs (but their assemblies are still connected to my project) i should take no (or minimal) perf hit.

This code will not do that.  We lose a MAJOR performance benefit here because we can no longer do this extremely quick syntactic check (which is the entire basis of FAR performance).

Reply:
So it's not clear to me what this API is used for precisely -- is it intended as a public form of a Navigate To API, or is this underneath some other feature we already have? I agree with @CyrusNajmabadi we need to be more careful here, or at least really convince ourselves we can't do something smarter.

I don't see any internal callers into Roslyn itself, but I'm doing that via a grep search because of all the overloads. So maybe I'm missing something?

Reply:
>  is it intended as a public form of a Navigate To API

Yes, this is what OmniSharp uses it for.